src/mixtures/mixturemodel.jl

```diff
@@ -359,15 +367,22 @@ function _mixlogpdf!(r::AbstractArray, d::AbstractMixtureModel, x)
     return r
 end
 
+function _mixgradlogpdf1(d::AbstractMixtureModel, x)
+    glp = insupport(d, x) ? sum(pi * pdf(d.components[i], x) .* gradlogpdf(d.components[i], x) for (i, pi) in enumerate(probs(d)) if (!iszero(pi) && !iszero(pdf(d.components[i], x)))) / pdf(d, x) : zero(x)
```

Comment:
I expect that in many cases this definition introduces a type instability (since generally the return types of the two branches will be different).

Reply:
I see. Makes sense. I will have another look when I get the chance. Thanks for the quick feedback.

Reply:
I believe the following would be an alternative to avoid type instability, right?

```
function Distributions.gradlogpdf(d::UnivariateMixture, x::Real)
          pdfx = pdf(d, x)
          glp = !iszero(pdfx) ? sum(pi * pdf(d.components[i], x) .* gradlogpdf(d.components[i], x) for (i, pi) in enumerate(probs(d)) if (!iszero(pi) && !iszero(pdf(d.components[i], x)))) / pdfx : pdfx
           return glp
       end
```